packages/blockchain-wallet-v4/src/network/api/earn/types.ts

```diff
@@ -44,12 +44,17 @@ export type EarnBondingDepositsResponseType = {
   bondingDeposits: Array<EarnBondingDepositsType>
   unbondingWithdrawals: Array<null>
 } | null
-export type EarnEligibleType = {
-  [key in CoinType]?: {
-    eligible: boolean
-    ineligibilityReason: 'KYC_TIER' | 'BLOCKED' | 'REGION' | 'UNSUPPORTED_COUNTRY_OR_STATE' | null
-  }
+
+// If user is eligible it will send {coin: eligibleType} otherwise it will send EligibleType only
+type EligibleType = {
+  eligible: boolean
+  ineligibilityReason: 'KYC_TIER' | 'BLOCKED' | 'REGION' | 'UNSUPPORTED_COUNTRY_OR_STATE' | null
```

Comment:
[]()packages/blockchain-wallet-v4/src/network/api/earn/types.ts**
```suggestion
  ineligibilityReason: 'KYC_TIER' | 'BLOCKED' | 'REGION' | 'UNSUPPORTED_COUNTRY_OR_STATE' | null
```
**